### PR TITLE
Add blacklist

### PR DESCRIPTION
**Pull request by yunfangjuan**

### CODE REVIEW

File: idseq_dag/util/m8.py [hunk 1]
```diff
@@ -73,10 +73,15 @@ def iterate_m8(m8_file, debug_caller=None, logging_interval=25000000):
             line_count, m8_file, debug_caller)
         log.write(msg)
 
+def read_file_into_set(file_name):
+    with open(file_name, 'r') as f:
+        S = set(x.rstrip() for x in f)
```
Comment (cdebourcy) [inline idseq_dag/util/m8.py]: or: `S = set(f.read().splitlines())`

File: idseq_dag/util/m8.py [hunk 2]
```diff
@@ -158,23 +166,29 @@ def get_lineage(accession_id):
...
         """Accumulate hits for summarizing hit information and specificity at
-        each taxonomy level
+        each taxonomy level.
+        Check if the taxid is blacklisted. If so, ignore the read
         """
+        blacklisted = False
         lineage_taxids = get_lineage(accession_id)
         for level, taxid_at_level in enumerate(lineage_taxids):
+            if taxid_at_level in blacklist_taxids:
+                blacklisted = True
```
[the remark this answers is not in the document]
Comment (cdebourcy) [reply idseq_dag/util/m8.py]: never mind

[the remark this answers is not in the document]
Comment (cdebourcy) [reply idseq_dag/util/m8.py]: I feel like it's hard to understand what's going on in the different places that "blacklisted" appears. Maybe a comment would help. Is it true that if one accession for a read is under a blacklisted taxid, then the entire read will be ignored?

[the remark this answers is not in the document]
Comment (yunfangjuan) [reply idseq_dag/util/m8.py]: I did have comment on the top.

**Pull request merged**